### PR TITLE
add levels and scale to camera

### PR DESCRIPTION
**Pull request by ruihan00**
- add complete level 3
- add scale to camera for higher levels
- fix bug in back button

### CODE REVIEW

File: SDPhysicsEngine/Sources/SDPhysicsEngine/Object/SDCameraObject.swift [hunk 1]
```diff
@@ -33,11 +35,11 @@ public class SDCameraObject: SDObject {
...
         var newPosition = player.position
-        if player.position.x < bounds.minX {
-            newPosition.x = bounds.minX
+        if player.position.x < bounds.minX + 200 {
+            newPosition.x = bounds.minX + 200
         }
-        if player.position.x > bounds.maxX {
-            newPosition.x = bounds.maxX
+        if player.position.x > bounds.maxX - 200 {
+            newPosition.x = bounds.maxX - 200
```
Comment (jasonqiu212) [inline SDPhysicsEngine/Sources/SDPhysicsEngine/Object/SDCameraObject.swift:42]: This is a bit hacky for #80. But we can fix this in the next sprint

**Review by jasonqiu212 — APPROVED** [review]
lgtm

**Comment by Junyi00** [comment]
image
it seems like the character starts outside the map

**Comment by jasonqiu212** [comment]
> image it seems like the character starts outside the map

Is this on the iPad Pro simulator? I think it's because of the issue I mentioned in #80

**Comment by ruihan00** [comment]
the root cause of the issue is prolly why there seems to be a offset even if the camera is set to the center of the screen lol, also scaling kinda impacts the  player position but we need it if not the full heigh of the map will be shown no matter wat @Junyi00 @jasonqiu212 , I will merge this in first, we can resolve in another pr

**Pull request merged**